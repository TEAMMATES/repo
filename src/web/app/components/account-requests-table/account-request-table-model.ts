/**
 * Model for the row entries in the account requests table.
 */
export interface AccountRequestTableRowModel {
    id: string;
    name: string;
    email: string;
    status: string;
    instituteAndCountry: string;
    createdAtText: string;
    registeredAtText: string;
    comments: string;
    registrationLink: string;
    showLinks: boolean;
}
