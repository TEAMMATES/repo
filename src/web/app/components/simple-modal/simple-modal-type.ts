/**
 * Represents the type of  the modal to be displayed.
 */
export enum SimpleModalType {
  /**
   * Warning modal.
   */
  WARNING,
  /**
   * Danger modal.
   */
  DANGER,
  /**
   * Info modal.
   */
  INFO,
  /**
   * Primary modal.
   */
  PRIMARY,
  /**
   * Neutral modal.
   */
  NEUTRAL,
}
