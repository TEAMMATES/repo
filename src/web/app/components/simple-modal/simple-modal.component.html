<div class="modal-header"
     [ngClass]="{
     'alert-danger' : type === SimpleModalType.DANGER,
     'alert-warning' : type === SimpleModalType.WARNING,
     'alert-info' : type === SimpleModalType.INFO,
     'bg-white' : type === SimpleModalType.NEUTRAL}">
  <div>
    <h5 class="modal-title">
      <i *ngIf="type === SimpleModalType.DANGER" class="fas fa-times-circle"></i>
      <i *ngIf="type === SimpleModalType.WARNING" class="fas fa-exclamation-circle"></i>
      <i *ngIf="type === SimpleModalType.INFO" class="fas fa-info-circle"></i>
      <span *ngIf="isHeaderTemplate" class="margin-left-5px">
        <ng-container *ngTemplateOutlet="header"></ng-container>
      </span>
      <span *ngIf="!isHeaderTemplate" class="margin-left-5px" [innerHTML]="header"></span>
    </h5>
  </div>
  <button type="button" class="close" tabindex="-1" (click)="activeModal.dismiss()">
    <i class="fas fa-times"></i>
  </button>
</div>
<div class="modal-body">
  <div *ngIf="isTemplate">
    <ng-container *ngTemplateOutlet="content; context: context"></ng-container>
  </div>
  <div *ngIf="!isTemplate" [innerHTML]="content"></div>
</div>
<div class="modal-footer" *ngIf="buttons.length">
  <button *ngFor="let button of buttons" type="button" tabindex="-1" class="btn {{ button.classes }}"
          (click)="activeModal.dismiss(); button.action && button.action.call();">
    {{ button.text }}
  </button>
</div>
