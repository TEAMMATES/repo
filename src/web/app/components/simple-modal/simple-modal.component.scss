.margin-left-5px {
  margin-left: 5px;
}

.progress-bar-container {
  margin-top: 10px;
  margin-bottom: 2px;
}
