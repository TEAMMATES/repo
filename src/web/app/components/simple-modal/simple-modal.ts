import { TemplateRef } from '@angular/core';

/**
 * Represents a button for the modal.
 */
export interface SimpleModalButton {
  text: string;
  classes?: string;
  action?: Function;
}

/**
 * The standard cancel button for modal.
 */
export const standardCancelButton: (text?: string) => SimpleModalButton = (text?: string) => {
  return {
    text: text || 'Cancel',
    classes: 'btn-light modal-btn-cancel',
  };
};

/**
 * The standard confirm button for modal.
 */
export const standardConfirmButton: (type: string, action: Function, text?: string) => SimpleModalButton =
    (type: string, action: Function, text?: string) => {
      return {
        action,
        text: text || 'Yes',
        classes: `${type} modal-btn-ok`,
      };
    };

/**
 * Parameters for modal.
 */
export interface SimpleModalOptions {
  type: SimpleModalType;
  header: string | TemplateRef<any>;
  content: string | TemplateRef<any>;
  context?: Record<string, any>;
  buttons?: SimpleModalButton[];
  options?: Record<string, string | boolean>;
}

/**
 * Represents the type of the modal to be displayed.
 */
export enum SimpleModalType {
  /**
   * Warning modal.
   */
  WARNING,
  /**
   * Danger modal.
   */
  DANGER,
  /**
   * Info modal.
   */
  INFO,
  /**
   * Neutral modal.
   */
  NEUTRAL,
}
