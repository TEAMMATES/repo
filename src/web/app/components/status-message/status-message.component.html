<div class="alert alert-{{ statusMessage.color }}" *ngFor="let statusMessage of messages">
  <button class="close" (click)="dismiss(statusMessage)">
    <span aria-hidden="true">&times;</span>
  </button>

  <!--TODO: Change this behavior when no status messages are returned in raw HTML-->
  <div style="padding-right: 30px;" [innerHtml]="statusMessage.message"></div>
</div>
