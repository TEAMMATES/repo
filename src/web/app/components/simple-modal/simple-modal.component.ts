import { Component, Input, TemplateRef } from '@angular/core';
import { Router } from '@angular/router';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
import { NavigationService } from '../../../services/navigation.service';
import { SimpleModalType } from './simple-modal-type';

/**
 * A component to display contents of confirmation modals.
 */
@Component({
  selector: 'tm-confirmation-modal',
  templateUrl: './simple-modal.component.html',
  styleUrls: ['./simple-modal.component.scss'],
})
export class SimpleModalComponent {

  // enum
  SimpleModalType: typeof SimpleModalType = SimpleModalType;

  @Input() header: string = '';
  @Input() content: string | TemplateRef<any> = '';
  @Input() type: SimpleModalType = SimpleModalType.NEUTRAL;
  @Input() isInformationOnly: boolean = false; // true will cause modal to only have 1 button
  @Input() confirmMessage: string = 'Yes'; // custom text message for confirm button
  @Input() cancelMessage: string = 'No, cancel the operation'; // custom text message for cancel button
  @Input() redirectionUrl: string = '';

  get isTemplate(): boolean {
    return this.content instanceof TemplateRef;
  }

  constructor(public activeModal: NgbActiveModal,
              private router: Router,
              private navigationService: NavigationService) { }

  redirect(): void {
    if (this.redirectionUrl) {
      this.navigationService.navigateByURL(this.router, this.redirectionUrl);
    }
  }

}
