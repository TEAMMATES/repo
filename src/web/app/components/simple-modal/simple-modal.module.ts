import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { AjaxLoadingModule } from '../ajax-loading/ajax-loading.module';
import { SimpleModalComponent } from './simple-modal.component';

/**
 * Module for the modal component
 */
@NgModule({
  declarations: [SimpleModalComponent],
  exports: [SimpleModalComponent],
  imports: [
    CommonModule,
    AjaxLoadingModule,
  ],
  entryComponents: [SimpleModalComponent],
})

export class SimpleModalModule { }
