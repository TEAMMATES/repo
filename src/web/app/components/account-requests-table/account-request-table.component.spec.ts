import { HttpClientTestingModule } from '@angular/common/http/testing';
import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { of, throwError } from 'rxjs';
import { AccountRequestTableRowModel } from './account-request-table-model';
import { AccountRequestTableComponent } from './account-request-table.component';
import { AccountRequestTableModule } from './account-request-table.module';
import { AccountService } from '../../../services/account.service';
import { SimpleModalService } from '../../../services/simple-modal.service';
import { StatusMessageService } from '../../../services/status-message.service';
import { createBuilder } from '../../../test-helpers/generic-builder';
import { createMockNgbModalRef } from '../../../test-helpers/mock-ngb-modal-ref';
import { AccountRequestStatus } from '../../../types/api-output';
import { SimpleModalType } from '../simple-modal/simple-modal-type';

describe('AccountRequestTableComponent', () => {
    let component: AccountRequestTableComponent;
    let fixture: ComponentFixture<AccountRequestTableComponent>;
    let accountService: AccountService;
    let statusMessageService: StatusMessageService;
    let simpleModalService: SimpleModalService;

    const accountRequestDetailsBuilder = createBuilder<AccountRequestTableRowModel>({
        id: '',
        email: '',
        name: '',
        instituteAndCountry: '',
        registrationLink: '',
        status: AccountRequestStatus.PENDING,
        comments: '',
        registeredAtText: '',
        createdAtText: '',
        showLinks: false,
    });

    const DEFAULT_ACCOUNT_REQUEST_PENDING = accountRequestDetailsBuilder
        .email('email')
        .name('name')
        .status(AccountRequestStatus.PENDING)
        .instituteAndCountry('institute')
        .createdAtText('Tue, 08 Feb 2022, 08:23 AM +00:00')
        .comments('comment');

    const DEFAULT_ACCOUNT_REQUEST_APPROVED = accountRequestDetailsBuilder
        .email('email')
        .name('name')
        .status(AccountRequestStatus.APPROVED)
        .registrationLink('registrationLink')
        .instituteAndCountry('institute')
        .createdAtText('createdTime')
        .comments('comment');

    const DEFAULT_ACCOUNT_REQUEST_REGISTERED = accountRequestDetailsBuilder
        .email('email')
        .name('name')
        .status(AccountRequestStatus.REGISTERED)
        .registrationLink('registrationLink')
        .instituteAndCountry('institute')
        .registeredAtText('registeredTime')
        .createdAtText('createdTime')
        .comments('comment');

    const DEFAULT_ACCOUNT_REQUEST_REJECTED = accountRequestDetailsBuilder
        .email('email')
        .name('name')
        .status(AccountRequestStatus.REJECTED)
        .registrationLink('registrationLink')
        .instituteAndCountry('institute')
        .createdAtText('createdTime')
        .comments('comment');

    const resetModalContent = `Are you sure you want to reset the account request for
        <strong>name</strong> with email <strong>email</strong> from
        <strong>institute</strong>?
        An email with the account registration link will also be sent to the instructor.`;
    const resetModalTitle = 'Reset account request for <strong>name</strong>?';
    const deleteModalContent = `Are you sure you want to <strong>delete</strong> the account request for
        <strong>name</strong> with email <strong>email</strong> from
        <strong>institute</strong>?`;
    const deleteModalTitle = 'Delete account request for <strong>name</strong>?';

    beforeEach(waitForAsync(() => {
        TestBed.configureTestingModule({
            declarations: [AccountRequestTableComponent],
            imports: [
                AccountRequestTableModule,
                BrowserAnimationsModule,
                HttpClientTestingModule,
            ],
            providers: [
                AccountService, SimpleModalService,
            ],
        }).compileComponents();
    }));

    beforeEach(() => {
        fixture = TestBed.createComponent(AccountRequestTableComponent);
        component = fixture.componentInstance;
        accountService = TestBed.inject(AccountService);
        statusMessageService = TestBed.inject(StatusMessageService);
        simpleModalService = TestBed.inject(SimpleModalService);
        fixture.detectChanges();
      });

      it('should create', () => {
        expect(component).toBeTruthy();
      });

      it('should snap with an expanded account requests table', () => {
        component.accountRequests = [
            DEFAULT_ACCOUNT_REQUEST_PENDING.build(),
        ];

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });

      it('should show account request links when expand all button clicked', () => {
        const accountRequestResult: AccountRequestTableRowModel = DEFAULT_ACCOUNT_REQUEST_APPROVED.build();
        component.accountRequests = [accountRequestResult];
        component.searchString = 'test';
        fixture.detectChanges();

        const button: any = fixture.debugElement.nativeElement.querySelector('#show-account-request-links');
        button.click();
        expect(component.accountRequests[0].showLinks).toEqual(true);
      });

      it('should display account requests with no reset or expand links button', () => {
        const accountRequestResults: AccountRequestTableRowModel[] = [
            DEFAULT_ACCOUNT_REQUEST_PENDING.build(),
            DEFAULT_ACCOUNT_REQUEST_PENDING.build(),
        ];

        component.accountRequests = accountRequestResults;

        expect(component.accountRequests.length).toEqual(2);
        expect(component.accountRequests).toEqual(accountRequestResults);
        expect(component.accountRequests[0].showLinks).toEqual(false);
        expect(component.accountRequests[1].showLinks).toEqual(false);

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });

      it('should display account requests with reset button and expandable links buttons',
      () => {
        const accountRequestResults: AccountRequestTableRowModel[] = [
            DEFAULT_ACCOUNT_REQUEST_APPROVED.build(),
            DEFAULT_ACCOUNT_REQUEST_REJECTED.build(),
        ];

        component.accountRequests = accountRequestResults;
        component.searchString = 'test';

        expect(component.accountRequests.length).toEqual(2);
        expect(component.accountRequests).toEqual(accountRequestResults);
        expect(component.accountRequests[0].showLinks).toEqual(false);
        expect(component.accountRequests[1].showLinks).toEqual(false);

        fixture.detectChanges();
        expect(fixture).toMatchSnapshot();
      });

      it('should show success message when deleting account request is successful', () => {
        component.accountRequests = [
            DEFAULT_ACCOUNT_REQUEST_PENDING.build(),
        ];
        fixture.detectChanges();

        const modalSpy = jest.spyOn(simpleModalService, 'openConfirmationModal').mockImplementation(() => {
            return createMockNgbModalRef({});
        });

        jest.spyOn(accountService, 'deleteAccountRequest').mockReturnValue(of({
          message: 'Account request successfully deleted.',
        }));

        const spyStatusMessageService: any = jest.spyOn(statusMessageService, 'showSuccessToast')
            .mockImplementation((args: string) => {
              expect(args).toEqual('Account request successfully deleted.');
            });

        const deleteButton: any = fixture.debugElement.nativeElement.querySelector('#delete-account-request-0');
        deleteButton.click();

        expect(spyStatusMessageService).toHaveBeenCalled();
        expect(modalSpy).toHaveBeenCalledTimes(1);
        expect(modalSpy).toHaveBeenCalledWith(deleteModalTitle, SimpleModalType.DANGER, deleteModalContent);
      });

      it('should show error message when deleting account request is unsuccessful', () => {
        component.accountRequests = [
            DEFAULT_ACCOUNT_REQUEST_PENDING.build(),
        ];

        fixture.detectChanges();

        const modalSpy = jest.spyOn(simpleModalService, 'openConfirmationModal').mockImplementation(() => {
            return createMockNgbModalRef({});
        });

        jest.spyOn(accountService, 'deleteAccountRequest').mockReturnValue(throwError(() => ({
          error: {
            message: 'This is the error message.',
          },
        })));

        const spyStatusMessageService: any = jest.spyOn(statusMessageService, 'showErrorToast')
          .mockImplementation((args: string) => {
            expect(args).toEqual('This is the error message.');
          });

        const deleteButton: any = fixture.debugElement.nativeElement.querySelector('#delete-account-request-0');
        deleteButton.click();

        expect(spyStatusMessageService).toHaveBeenCalled();
        expect(modalSpy).toHaveBeenCalledTimes(1);
        expect(modalSpy).toHaveBeenCalledWith(deleteModalTitle, SimpleModalType.DANGER, deleteModalContent);
      });

      it('should show success message when resetting account request is successful', () => {
        component.accountRequests = [
            DEFAULT_ACCOUNT_REQUEST_REGISTERED.build(),
        ];

        component.searchString = 'test';
        fixture.detectChanges();

        const modalSpy = jest.spyOn(simpleModalService, 'openConfirmationModal').mockImplementation(() => {
            return createMockNgbModalRef({});
        });

        jest.spyOn(accountService, 'resetAccountRequest').mockReturnValue(of({
          joinLink: 'joinlink',
        }));

        const spyStatusMessageService = jest.spyOn(statusMessageService, 'showSuccessToast')
          .mockImplementation((args: string) => {
            expect(args)
                .toEqual('Reset successful. An email has been sent to email.');
          });

        const resetButton = fixture.debugElement.nativeElement.querySelector('#reset-account-request-0');
        resetButton.click();

        expect(spyStatusMessageService).toHaveBeenCalled();
        expect(modalSpy).toHaveBeenCalledTimes(1);
        expect(modalSpy).toHaveBeenCalledWith(resetModalTitle, SimpleModalType.WARNING, resetModalContent);
      });

      it('should show error message when resetting account request is unsuccessful', () => {
        component.accountRequests = [
            DEFAULT_ACCOUNT_REQUEST_REGISTERED.build(),
        ];
        component.searchString = 'test';
        fixture.detectChanges();

        const modalSpy = jest.spyOn(simpleModalService, 'openConfirmationModal').mockImplementation(() => {
          return createMockNgbModalRef({});
        });

        jest.spyOn(accountService, 'resetAccountRequest').mockReturnValue(throwError(() => ({
          error: {
            message: 'This is the error message.',
          },
        })));

        const spyStatusMessageService = jest.spyOn(statusMessageService, 'showErrorToast')
          .mockImplementation((args: string) => {
            expect(args).toEqual('This is the error message.');
          });

        const resetButton = fixture.debugElement.nativeElement.querySelector('#reset-account-request-0');
        resetButton.click();

        expect(spyStatusMessageService).toHaveBeenCalled();
        expect(modalSpy).toHaveBeenCalledTimes(1);
        expect(modalSpy).toHaveBeenCalledWith(resetModalTitle, SimpleModalType.WARNING, resetModalContent);
      });

      it('should display comment modal', () => {
        const accountRequestResults: AccountRequestTableRowModel[] = [
            DEFAULT_ACCOUNT_REQUEST_PENDING.build(),
        ];

        component.accountRequests = accountRequestResults;
        fixture.detectChanges();

        const modalSpy = jest.spyOn(simpleModalService, 'openInformationModal')
            .mockReturnValue(createMockNgbModalRef());

        const viewCommentButton: any = fixture.debugElement.nativeElement.querySelector('#view-account-request-0');
        viewCommentButton.click();
        expect(modalSpy).toHaveBeenCalledTimes(1);
        expect(modalSpy).toHaveBeenCalledWith('Comments for <strong>name</strong> Request',
            SimpleModalType.INFO, '<strong>Comment:</strong> comment');
      });
});
