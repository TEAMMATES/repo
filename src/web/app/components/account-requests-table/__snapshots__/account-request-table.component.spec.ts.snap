// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountRequestTableComponent should display account requests with no reset or expand links button 1`] = `
<tm-account-request-table
  accountRequests={[Function Array]}
  accountService={[Function AccountService]}
  searchString=""
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <strong>
        Pending Account Requests
      </strong>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped data-table"
        id="search-table-account-request"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Status
            </th>
            <th>
              Institute, Country
            </th>
            <th>
              Created At
            </th>
            <th>
              Comments
            </th>
            <th>
              Options
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              name
            </td>
            <td>
              email
            </td>
            <td>
              REGISTERED
            </td>
            <td>
              institute
            </td>
            <td
              id="timestamp-box"
            >
              createdTime
            </td>
            <td>
              <div
                id="comment-box"
              >
                 comment 
              </div>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-center gap-2"
              >
                <div
                  class="ngb-tooltip-class"
                >
                  <a
                    href="javascript:;"
                    id="delete-account-request-0"
                  >
                    <i
                      class="fa-solid fa-trash"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="javascript:;"
                    id="view-account-request-0"
                  >
                    <i
                      class="fa-solid fa-eye"
                    />
                  </a>
                </div>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              name
            </td>
            <td>
              email
            </td>
            <td>
              REGISTERED
            </td>
            <td>
              institute
            </td>
            <td
              id="timestamp-box"
            >
              createdTime
            </td>
            <td>
              <div
                id="comment-box"
              >
                 comment 
              </div>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-center gap-2"
              >
                <div
                  class="ngb-tooltip-class"
                >
                  <a
                    href="javascript:;"
                    id="delete-account-request-1"
                  >
                    <i
                      class="fa-solid fa-trash"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="javascript:;"
                    id="view-account-request-1"
                  >
                    <i
                      class="fa-solid fa-eye"
                    />
                  </a>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</tm-account-request-table>
`;

exports[`AccountRequestTableComponent should display account requests with reset button and expandable links buttons 1`] = `
<tm-account-request-table
  accountRequests={[Function Array]}
  accountService={[Function AccountService]}
  searchString={[Function String]}
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <div>
        <strong>
          Account Requests Found
        </strong>
      </div>
      <div
        class="card-header-btn-toolbar"
      >
        <button
          class="btn btn-light btn-sm"
          id="show-account-request-links"
          style="margin-right: 10px;"
          type="button"
        >
          Expand All
        </button>
        <button
          class="btn btn-light btn-sm"
          id="hide-account-request-links"
          type="button"
        >
          Collapse All
        </button>
      </div>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped data-table"
        id="search-table-account-request"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Status
            </th>
            <th>
              Institute, Country
            </th>
            <th>
              Created At
            </th>
            <th>
              Registered At
            </th>
            <th>
              Comments
            </th>
            <th>
              Options
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              name
            </td>
            <td>
              email
            </td>
            <td>
              REGISTERED
            </td>
            <td>
              institute
            </td>
            <td
              id="timestamp-box"
            >
              createdTime
            </td>
            <td
              id="timestamp-box"
            >
              registeredTime
            </td>
            <td>
              <div
                id="comment-box"
              >
                 comment 
              </div>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-center gap-2"
              >
                <div
                  class="ngb-tooltip-class"
                >
                  <a
                    href="javascript:;"
                    id="delete-account-request-0"
                  >
                    <i
                      class="fa-solid fa-trash"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="javascript:;"
                    id="view-account-request-0"
                  >
                    <i
                      class="fa-solid fa-eye"
                    />
                  </a>
                </div>
                <div
                  class="ngb-tooltip-class"
                >
                  <button
                    class="btn btn-primary"
                    id="reset-account-request-0"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </td>
          </tr>
          <tr>
            <td>
              name
            </td>
            <td>
              email
            </td>
            <td>
              REGISTERED
            </td>
            <td>
              institute
            </td>
            <td
              id="timestamp-box"
            >
              createdTime
            </td>
            <td
              id="timestamp-box"
            >
              registeredTime
            </td>
            <td>
              <div
                id="comment-box"
              >
                 comment 
              </div>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-center gap-2"
              >
                <div
                  class="ngb-tooltip-class"
                >
                  <a
                    href="javascript:;"
                    id="delete-account-request-1"
                  >
                    <i
                      class="fa-solid fa-trash"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="javascript:;"
                    id="view-account-request-1"
                  >
                    <i
                      class="fa-solid fa-eye"
                    />
                  </a>
                </div>
                <div
                  class="ngb-tooltip-class"
                >
                  <button
                    class="btn btn-primary"
                    id="reset-account-request-1"
                  >
                    Reset
                  </button>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</tm-account-request-table>
`;

exports[`AccountRequestTableComponent should snap with an expanded account requests table 1`] = `
<tm-account-request-table
  accountRequests={[Function Array]}
  accountService={[Function AccountService]}
  searchString=""
  simpleModalService={[Function SimpleModalService]}
  statusMessageService={[Function StatusMessageService]}
>
  <div
    class="card bg-light top-padded"
  >
    <div
      class="card-header bg-primary text-white"
    >
      <strong>
        Pending Account Requests
      </strong>
    </div>
    <div
      class="table-responsive"
    >
      <table
        class="table table-striped data-table"
        id="search-table-account-request"
      >
        <thead>
          <tr>
            <th>
              Name
            </th>
            <th>
              Email
            </th>
            <th>
              Status
            </th>
            <th>
              Institute, Country
            </th>
            <th>
              Created At
            </th>
            <th>
              Comments
            </th>
            <th>
              Options
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>
              name
            </td>
            <td>
              email
            </td>
            <td>
              REGISTERED
            </td>
            <td>
              institute
            </td>
            <td
              id="timestamp-box"
            >
              createdTime
            </td>
            <td>
              <div
                id="comment-box"
              >
                 comment 
              </div>
            </td>
            <td
              class="align-middle"
            >
              <div
                class="d-flex flex-row align-items-center justify-content-center gap-2"
              >
                <div
                  class="ngb-tooltip-class"
                >
                  <a
                    href="javascript:;"
                    id="delete-account-request-0"
                  >
                    <i
                      class="fa-solid fa-trash"
                    />
                  </a>
                </div>
                <div>
                  <a
                    href="javascript:;"
                    id="view-account-request-0"
                  >
                    <i
                      class="fa-solid fa-eye"
                    />
                  </a>
                </div>
              </div>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</tm-account-request-table>
`;
